online-shopping-website/backend/src/routes/users.ts

```diff
@@ -67,8 +67,12 @@ userRouter.post('/update', async (req: Request, res: Response) => { // updates a
     if (usr === null || usr === undefined) {
       throw new Error(`Authentication is invalid`);
     }
+    const match = await bcrypt.compare(req.body.oldPassword, usr.password);
+    if (!match){
+      throw new Error(`Password is incorrect`);
+    }
     let encrypted_password: string | undefined;
-    if (req.body.password !== undefined) {
+    if (req.body.newPassword !== undefined) {
       //new password
       encrypted_password = await bcrypt.hash(req.body.password, 5);
```

Comment:
This tries to hash `req.body.password` instead of `req.body.newPassword` is this correct?

Reply:
I get what you mean. But on line 77, it does `bcrypt.hash(req.body.password, 5)`. As per the wiki, you take in `oldPassword` as a key, not `password`. I just want to make sure I'm sending the correct body key/value pairs. Doesn't `password` no longer exist in the request body?

Reply:
Yeah thats my bad, i reverted back to `password` so that there's no confusion